homeassistant/components/zwave_js/humidifier.py

```diff
@@ -0,0 +1,214 @@
+"""Representation of Z-Wave thermostats."""
+from __future__ import annotations
+
+from typing import Any
+
+from zwave_js_server.client import Client as ZwaveClient
+from zwave_js_server.const import CommandClass
+from zwave_js_server.const.command_class.humidity_control import (
+    HUMIDITY_CONTROL_MODE_PROPERTY,
+    HUMIDITY_CONTROL_SETPOINT_PROPERTY,
+    HumidityControlMode,
+    HumidityControlSetpointType,
+)
+from zwave_js_server.model.value import Value as ZwaveValue
+
+from homeassistant.components.humidifier import HumidifierDeviceClass, HumidifierEntity
+from homeassistant.components.humidifier.const import (
+    DEFAULT_MAX_HUMIDITY,
+    DEFAULT_MIN_HUMIDITY,
+    DOMAIN as HUMIDIFIER_DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DATA_CLIENT, DOMAIN
+from .discovery import ZwaveDiscoveryInfo
+from .entity import ZWaveBaseEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Z-Wave humidifier from config entry."""
+    client: ZwaveClient = hass.data[DOMAIN][config_entry.entry_id][DATA_CLIENT]
+
+    @callback
+    def async_add_humidifier(info: ZwaveDiscoveryInfo) -> None:
+        """Add Z-Wave Humidifier."""
+        entities: list[ZWaveBaseEntity] = []
+
+        if (
+            str(HumidityControlMode.HUMIDIFY.value)
+            in info.primary_value.metadata.states
+        ):
+            entities.append(ZWaveHumidifier(config_entry, client, info))
+
+        if (
+            str(HumidityControlMode.DEHUMIDIFY.value)
+            in info.primary_value.metadata.states
+        ):
+            entities.append(ZWaveDehumidifier(config_entry, client, info))
+
+        async_add_entities(entities)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(
+            hass,
+            f"{DOMAIN}_{config_entry.entry_id}_add_{HUMIDIFIER_DOMAIN}",
+            async_add_humidifier,
+        )
+    )
+
+
+class ZWaveBaseHumidifier(ZWaveBaseEntity, HumidifierEntity):
+    """Representation of a Z-Wave Humidifier or Dehumidifier."""
+
+    _current_mode: ZwaveValue
+    _setpoint: ZwaveValue | None = None
+
+    _on_mode = HumidityControlMode
+    _inverse_mode = HumidityControlMode
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        client: ZwaveClient,
+        info: ZwaveDiscoveryInfo,
+    ) -> None:
+        """Initialize thermostat."""
+        super().__init__(config_entry, client, info)
+
+        self._attr_name = f"{self._attr_name} {self.device_class}"
+        self._attr_unique_id = f"{self._attr_unique_id}_{self.device_class}"
```

Comment:
this turned out to be necessary, otherwise both the humidifier and the de-humidifier will get the same unique_id and name, and only the first entity will be accessible

Reply:
Don't use the device class. Define a string that the integration is in control of and use that as suffix.